retrieval/target.go

```diff
@@ -196,11 +198,14 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	if cfg.BasicAuth != nil {
 		t.url.User = url.UserPassword(cfg.BasicAuth.Username, cfg.BasicAuth.Password)
 	}
+	t.url.RawQuery = cfg.Params.Encode()
```

Comment:
For the usecases I'm thinking of (e.g. passing instance down to snmp exporter) they'd be tied very closely to this code, so I don't think the restriction will be a problem in practice.